src/me/drton/jmavsim/UDPMavLinkPort.java

```diff
@@ -49,20 +49,37 @@ public void setDebug(boolean debug) {
         this.debug = debug;
     }
 
-    public void setup(String peerAddress, int peerPort) throws UnknownHostException, IOException {
+    public void setupClient(String peerAddress, int peerPort) throws UnknownHostException, IOException {
         this.peerPort = new InetSocketAddress(peerAddress, peerPort);
         this.bindPort = new InetSocketAddress("0.0.0.0", 0);
         if (debug) {
             System.out.println("peerAddress: " + this.peerPort.toString() + ", bindAddress: " +
                                this.bindPort.toString());
         }
+
+        channel = DatagramChannel.open();
+        channel.socket().bind(bindPort);
+        channel.configureBlocking(false);
+        channel.connect(this.peerPort);
     }
 
-    public void open() throws IOException {
+    public void setupHost(int peerPort) throws UnknownHostException, IOException {
+        //this.peerPort = new InetSocketAddress(peerAddress, 0);
+        this.bindPort = new InetSocketAddress("0.0.0.0", peerPort);
+        if (debug) {
+            System.out.println("peerAddress: " + this.peerPort.toString() + ", bindAddress: " +
+                               this.bindPort.toString());
```

Comment:
Since we have a blocking call below, it might be worth printing this in any case, so the user sees what we are waiting for (in case no data comes in).

Reply:
Good point, added 2590b24:

```
waiting for first message from: /0.0.0.0:14550
received first message from: /10.41.1.1:37905
```